homeassistant/components/watchdog_file_watcher.py

```diff
@@ -0,0 +1,75 @@
+"""
+Component for monitoring activity on a folder.
+"""
+import asyncio
+import logging
+import voluptuous as vol
+from watchdog.observers import Observer
+from watchdog.events import PatternMatchingEventHandler
+
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+#DEPENDENCIES = ['watchdog']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_PATH = 'folder'
+DOMAIN = "watchdog_file_watcher"
+EVENT_TYPE = "event_type"
+SRC_PATH = "src_path"
+PATTERNS = ["*.txt", "*.py", "*.md", "*.jpg", "*.png"]
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_PATH): cv.isdir})
+}, extra=vol.ALLOW_EXTRA)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Set up the folder watcher."""
+    conf = config.get(DOMAIN)
+    path = conf.get(CONF_PATH)
+    if not hass.config.is_allowed_path(path):
+        _LOGGER.error("folder %s is not valid or allowed", path)
+    else:
+        Watcher(path, hass)
+    return True
+
+
+class Watcher(Entity):
+    """Class for starting Watchdog."""
+    def __init__(self, path, hass):
+        self._observer = Observer()
+        self._observer.schedule(
+            MyHandler(hass),
+            path,
+            recursive=True)
+        self._observer.start()
+
+
+class MyHandler(PatternMatchingEventHandler):
+    patterns = PATTERNS
+
+    def __init__(self, hass):
+        super().__init__()
+        self.hass = hass
+
+    def process(self, event):
+        """Process the Watchdog event."""
+        self.hass.bus.fire(
+            DOMAIN, {
+                EVENT_TYPE: event.event_type,
```

Comment:
I think you need to specify the event_type as a positional arg/kwarg. [`fire` signature](https://github.com/home-assistant/home-assistant/blob/e7b84432f94ffdd0d1e05550e8f2d506f23ae526/homeassistant/core.py#L380):

```python
def fire(self, event_type: str, event_data=None, origin=EventOrigin.local):
```

Also, I don't think we should be directly using the `event_type`s from watchdog, we should at least translate them to `EVENT_*` constants even if they end up being the same string. I mean it probably won't happen, but what if watchdog changes their event types.

Reply:
Well Otto, that's where Rob is one step ahead of you 😉 He is actually firing `watchdog_file_watcher` events. The subtype is the watchdog type.

I am not sure if we would want to have 1 event with sub event types or have each watchdog event type be it's own event in Home Assistant. I kinda like how it is now.

Reply:
I want automations to trigger on particular event types and not make this too complicated, I also like how it is now